src/attestation/Attestation.chain.ts

```diff
@@ -37,9 +36,17 @@ export async function store(
   return blockchain.submitTx(identity, tx)
 }
 
-function decode(encoded: QueryResult, claimHash: string): Attestation | null {
-  if (encoded && encoded.encodedLength) {
+function decode(
+  encoded: Option<Tuple> | Tuple,
+  claimHash: string
+): Attestation | null {
+  if (
+    encoded instanceof Option ||
```

Comment:
Since this check is done three times, I would recommend to move it to a function like `isNotEmpty`. Also, I would add a comment why it suffices to check for indices 0 and 1 of encoded not to be empty for better clarity.

Reply:
There is nothing systematic about checking indices `0` and `1` here; they just happen in these specific cases to be of types that would give away if the whole thing was invalid (we always expect a hash and an account id). I should add comments.
In fact, I think these checks are superfluous; at least they are if any one of the following assumptions are true:
1. A valid Hash will always have at least one non-zero byte when represented as a byte array
2. A valid AccountId will always have at least one non-zero byte when represented as a byte array

Tuples are just concatenations of the individual encoded values (https://substrate.dev/docs/en/conceptual/core/codec), so if either byte representation can tell us whether we have empty values or not, we'd see it in the Tuple's byte representation

Reply:
I'm tempted to remove this last part but not 100 % sure the checking of non-null byte is reliable. What do you think I should do?

Reply:
I would try to break it in a test and check whether the second check for indices 0 and 1 is really needed. Secondly, I would check your assumptions, e.g. by checking, whether you can create a SHA-256 hash and AccountId from a 0-byte array.

Reply:
Well, you can, and that is the issue:
not so much for a hash, where the string representation is just the hex:
`0x0000000000000000000000000000000000000000000000000000000000000000`
but quite prominently for an AccountId, as this is the string representation of a 0-byte array:
`5C4hrfjw9DjXZTzV3MwzrrAr9P1MJhSrvWGWqi1eSuyUpnhM` 
I guess your questions concerns whether this could be any actual account's address, but this I just don't know... 
I do know that both are basically just byte arrays of fixed length (`Ua8Fixed`) for which the `isEmpty` property is defined as "Returns true if the type wraps an empty/default all-0 value".